src/api/v1alpha1/edgeconnect/properties.go

```diff
@@ -0,0 +1,38 @@
+package edgeconnect
+
+import (
+	"fmt"
+
+	"github.com/Dynatrace/dynatrace-operator/src/api"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const defaultEdgeConnectRepository = "docker.io/dynatrace/edgeconnect"
+
+func (edgeConnect *EdgeConnect) Image() string {
+	repository := defaultEdgeConnectRepository
+	tag := api.LatestTag
+
+	if edgeConnect.Spec.ImageRef.Repository != "" {
+		repository = edgeConnect.Spec.ImageRef.Repository
+	}
+	if edgeConnect.Spec.ImageRef.Tag != "" {
+		tag = edgeConnect.Spec.ImageRef.Tag
+	}
+
+	return fmt.Sprintf("%s:%s", repository, tag)
+}
+
+func (edgeConnect *EdgeConnect) IsCustomImage() bool {
+	return edgeConnect.Spec.ImageRef.Repository != ""
+}
+
+func (edgeConnect *EdgeConnect) PullSecretWithoutData() corev1.Secret {
```

Comment:
nitpick: add a comment so we know what it is meant to be used for ("why does method exist")